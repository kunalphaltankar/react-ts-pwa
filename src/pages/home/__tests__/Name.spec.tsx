import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Name from "../components/name";

import candidateDetails from "./__mocks__/candidateDetails.json";

const element = (
  <Name
    name={candidateDetails.name}
    country={candidateDetails.country}
    designation={candidateDetails.designation}
    profilePhotoUri={candidateDetails.profilePhotoUri}
    state={candidateDetails.state}
  />
);

describe("Name", () => {
  it("should match the snapshot", () => {
    const { container } = render(element);

    expect(container).toMatchSnapshot();
  });

  it("should have name as header", () => {
    render(element);
    const nameHeader = screen.getByRole("heading", {
      name: candidateDetails.name,
    });

    expect(nameHeader).toBeInTheDocument();
  });

  it("should contain country, state, designation, profile photo.", () => {
    render(element);
    [
      candidateDetails.country,
      candidateDetails.state,
      candidateDetails.state,
    ].forEach((detail) => {
      expect(screen.getByText(detail)).toBeInTheDocument();
    });
    const profileImage = screen.getByAltText("Profile Image");
    expect(profileImage).toBeInTheDocument();
  });

  it("should contain Facebook, Twitter, LinkedIn, GitHub buttons", () => {
    render(element);
    [
      "Facebook profile",
      "Twitter profile",
      "LinkedIn profile",
      "GitHub profile",
    ].forEach((button) => {
      expect(screen.getByRole("button", { name: button })).toBeInTheDocument();
    });
  });
});
