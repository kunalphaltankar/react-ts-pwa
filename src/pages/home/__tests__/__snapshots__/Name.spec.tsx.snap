// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name should match the snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-1qxw92a-MuiStack-root"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular css-7pbt6e-MuiAvatar-root"
    >
      <img
        alt="Profile Image"
        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="https://picsum.photos/id/237/200"
      />
    </div>
    <div
      class="MuiStack-root css-nen11g-MuiStack-root"
    >
      <div
        class="MuiStack-root css-1qxw92a-MuiStack-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          William Sample
        </h6>
        <button
          aria-label="Facebook profile"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="FacebookIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Twitter profile"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="TwitterIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="LinkedIn profile"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="LinkedInIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="GitHub profile"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="GitHubIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiStack-root css-1hezniw-MuiStack-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          Senior Product Manager
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          United States
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          Dallas
        </p>
      </div>
    </div>
  </div>
</div>
`;
